src/app/txRunner.js

```diff
@@ -65,25 +65,28 @@ TxRunner.prototype.execute = function () {
         })
       })
     } else {
-      self.web3.eth.estimateGas(tx, function (err, resp) {
+      self.web3.eth.estimateGas(tx, function (err, gasEstimation) {
         if (err) {
-          return callback(err, resp)
+          return callback(err, gasEstimation)
         }
-
-        if (resp > gasLimit) {
-          return callback('Gas required exceeds limit: ' + resp)
-        }
-
-        tx.gas = resp
-
-        var sendTransaction = self.personalMode ? self.web3.personal.unlockAccountAndSendTransaction : self.web3.eth.sendTransaction
-
-        sendTransaction(tx, function (err, resp) {
+        self.web3.eth.getBlock('latest', function (err, block) {
           if (err) {
-            return callback(err, resp)
-          }
+            return callback(err)
+          } else {
+            var blockGasLimit = Math.floor(block.gasLimit - block.gasLimit / 1024)
+            tx.gas = blockGasLimit < gasEstimation ? blockGasLimit : gasEstimation
```

Comment:
The original problem here was that go-ethereum returns an estimate that is higher than the block gas limit if an exception occurs and this PR hopefully fixes it. Note that you would want to execute failing transactions exactly for debugging where the exception comes from.